### PR TITLE
Added custom user model

### LINKED ISSUE #1: Custom User Model Support Request
I would like to request the addition of a custom user model to the djangoessential library. The custom user model should allow developers to easily extend or modify the default user fields. Here are some suggested features:

- Support for additional fields like `phone_number`, `date_of_birth`, etc.
- Integration with Django's built-in authentication and permissions system.
- Easy migration from Django's default user model.

This would enhance the flexibility and usability of the library for developers working on various projects.

### LINKED ISSUE #2: Documentation for Custom User Model Implementation
With the addition of a custom user model to the djangoessential library, it would be highly beneficial to provide comprehensive documentation. 

The documentation should cover the following aspects:

- How to enable and configure the custom user model in a Django project.
- Examples of extending the user model with additional fields.
- Integration steps with Django's authentication system.
- Migration guide for existing projects using Django's default user model.
- Best practices for using the custom user model in various scenarios.

Having clear and detailed documentation will help developers implement and utilize the custom user model more effectively in their projects.

This issue is related to the custom user model implementation request (#1).

### PR DESCRIPTION
**Pull request by crabcatcher0**
If this might help..

### CODE REVIEW

File: djangoessentials/djangoessentials.py [hunk 1]
```diff
@@ -15,3 +17,40 @@ class MyS3Storage(S3Boto3Storage):
...
+
+
+class CustomUser(AbstractBaseUser, PermissionsMixin):
+    username = models.CharField(max_length=150, unique=True)
+    phone_number = models.CharField(max_length=15, blank=True, null=True)
+    date_of_birth = models.DateField(blank=True, null=True)
+    is_active = models.BooleanField(default=True)
+    is_staff = models.BooleanField(default=False)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
```
Comment (CoderMungan) [inline djangoessentials/djangoessentials.py:46]: Thank you for collabration. Every part amazing. As u know we have a `TimeBasedStampModel` so you can fix the with it (with inheritance).

The other part looks like great for me!

File: djangoessentials/djangoessentials.py [hunk 2]
```diff
@@ -15,3 +17,40 @@ class MyS3Storage(S3Boto3Storage):
...
+            raise ValueError('The Username field must be set')
+        user = self.model(username=username, **extra_fields)
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+    
+    def create_superuser(self, username, password=None, **extra_fields):
+        extra_fields.setdefault('is_staff', True)
+        extra_fields.setdefault('is_superuser', True)
+        return self.create_user(username, password, **extra_fields)
```
Comment (CoderMungan) [inline djangoessentials/djangoessentials.py:36]: Good to have!

**Comment by CoderMungan** [comment]
Thank you for collabration <3. We are glad to see you in this OSS. This PR linked with #2! U can sign your self there too.
Thank you!

File: djangoessentials/djangoessentials.py [hunk 3]
```diff
@@ -15,3 +17,38 @@ class MyS3Storage(S3Boto3Storage):
...
+
+
+class CustomUser(AbstractBaseUser, PermissionsMixin):
```
Comment (CoderMungan) [inline djangoessentials/djangoessentials.py:39]: its will be great to add this thing too. `class CustomUser(AbstractBaseUser, PermissionsMixin, TimeBasedStampModel):` @crabcatcher0

**Comment by crabcatcher0** [comment]
both class will inherit from TimpeStampModel now

File: djangoessentials/djangoessentials.py (hunk 3 above)
Comment (crabcatcher0) [reply djangoessentials/djangoessentials.py]: Thank you for guiding.

File: djangoessentials/djangoessentials.py [hunk 4]
```diff
@@ -15,3 +17,38 @@ class MyS3Storage(S3Boto3Storage):
     location = 'media/'  # S3'te files alt dir
     file_overwrite = False  # if have a same name about the file cant overwrite
     default_acl = 'public-read' # just can read permission
+
+
+
+#CustomUser Model
+class CustomUserManager(BaseUserManager, TimeBasedStampModel):
```
Comment (CoderMungan) [inline djangoessentials/djangoessentials.py:24]: in here we dont need it to have `TimeBasedStampModel` bro <3

File: djangoessentials/djangoessentials.py [hunk 5]
```diff
@@ -15,3 +17,38 @@ class MyS3Storage(S3Boto3Storage):
...
+   
+
+    objects = CustomUserManager()
+
+    USERNAME_FIELD = 'username'
+    REQUIRED_FIELDS = []
+
+    def __str__(self):
+        return self.username
+
```
Comment (CoderMungan) [inline djangoessentials/djangoessentials.py:54]: Looks so great!

**Review by CoderMungan — APPROVED** [review]
LGTM! Congrats for collaborator <3

File: djangoessentials/djangoessentials.py (hunk 4 above)
Comment (crabcatcher0) [reply djangoessentials/djangoessentials.py]: Thank you. I thought it would be helpful for admin too.

**Comment by CoderMungan** [comment]
If u have a pypi account can u gimme your username. I will add to you from pypi too for maintainers. Thank you so much @crabcatcher0 <3

**Comment by CoderMungan** [comment]
> > If u have a pypi account can u gimme your username. I will add to you from pypi too for maintainers. Thank you so much @crabcatcher0 <3
> 
> I lost my old account i created it now: crabcatcher0

i send it invite this account `crabcatcher0`! Thank you <3

**Comment by crabcatcher0** [comment]
Thank You !!

**Pull request merged**